flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/combine/CombineRecordReader.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.combine;
+
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.runtime.typeutils.RowDataSerializer;
+import org.apache.flink.table.storage.filestore.KeyValue;
+import org.apache.flink.table.storage.filestore.utils.RecordIterator;
+import org.apache.flink.table.storage.filestore.utils.RecordReader;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.Comparator;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * Abstract combine {@link RecordReader}. Combine key-values to key-value. The input {@link
+ * RecordReader} is sorted by key and sequence number.
+ *
+ * <p>This reader needs to hold at least two record objects at the same time to determine if
+ * previous one can be returned. So the reader only accepts wrapped reader with {@link
```

Comment:
> `This reader needs to hold at least two record objects at the same time to determine if previous one can be returned`

I was wondering the combined record may not exactly the same as the previous or current one, am I right?
`This reader needs to hold at least two record objects at the same time to perform a combined process`?

Reply:
Yes, the comment is a misunderstanding.